pkg/webhook/managedresource/managedresource_validating_webhook_test.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package managedresource
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	admissionv1 "k8s.io/api/admission/v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func Test_managedResourceValidzaator_Handle(t *testing.T) {
+	const fleet1p = "fleet1p"
+	validator := &managedResourceValidator{
+		whiteListedUsers: []string{fleet1p},
+	}
+
+	tests := []struct {
+		name           string
+		username       string
+		operation      admissionv1.Operation
+		oldLabels      map[string]string
+		oldAnnotations map[string]string
+		newLabels      map[string]string
+		newAnnotations map[string]string
+		expectedResp   admission.Response
+		modReq         func(*admission.Request)
+	}{
+		{
+			name:           "allowed when not managed by arm",
+			operation:      admissionv1.Update,
+			oldLabels:      map[string]string{"foo": "bar"},
+			oldAnnotations: map[string]string{"baz": "qux"},
+			newLabels:      map[string]string{"foo": "bar"},
+			newAnnotations: map[string]string{"baz": "qux"},
+			expectedResp:   admission.Allowed(""),
+		},
+		{
+			name:         "denied - error on getLabelsAndAnnotations failure",
+			operation:    admissionv1.Create,
+			expectedResp: admission.Errored(http.StatusInternalServerError, fmt.Errorf("object does not implement the Object interfaces")),
+			modReq: func(req *admission.Request) {
+				req.Object = runtime.RawExtension{Object: nil}
+			},
+		},
+		{
+			name:           "denied - managed by arm in labels, not whitelisted",
+			operation:      admissionv1.Create,
+			oldLabels:      nil,
+			oldAnnotations: nil,
+			newLabels:      map[string]string{managedByArmKey: managedByArmValue},
+			newAnnotations: nil,
+			expectedResp:   admission.Denied(fmt.Sprintf(resourceDeniedFormat, metav1.GroupVersionKind{Kind: "TestKind"}, "test-resource", "default")),
+		},
+		{
+			name:           "denied - managed by arm in annotations, not whitelisted",
```

Comment:
This case is a some user adding ARM annotations to a unmanaged resource right ?

Does ARM take ownership of unmanaged resources after creation or does it add labels/annotation on creation ?

Reply:
It's at the creation.  In fact, it just applies (as in `kubectl apply`) it whether it's a create or update operation.  When taking over, the resources will be over written.

And to answer your question, yes, we prevent users from adding the labels or annotations that indicate it's a managed resource.  Everything will have to go through ARM.

Reply:
If the labels/annotations are added during creation then no user including admin users should be able to update ARM managed labels/annotations right ?